fix: clean up warning for wildcard syntax usage (#329)
openfga/openfga.dev: docs/content/modeling/public-access.mdx

- A <ProductConcept section="what-is-a-relation" linkName="Relation" />: is a string defined in the type definition of an authorization model that defines the possibility of a relationship between an object of the same type as the type definition and a user in the system
- An <ProductConcept section="what-is-an-object" linkName="Object" />: represents an entity in the system. Users' relationships to it can be define through relationship tuples and the authorization model
- A <ProductConcept section="what-is-a-relationship-tuple" linkName="Relationship Tuple" />: a grouping consisting of a user, a relation and an object stored in <ProductName format={ProductNameFormat.ShortForm}/>

> - A <ProductConcept section="what-is-a-type-bound-public-access" linkName="Type Bound Public Access" />: is a special <ProductName format={ProductNameFormat.ShortForm}/> concept (represented by `<type>:*`) can be used in relationship tuples to represent every user

Reviewer: ```suggestion
- A <ProductConcept section="what-is-a-type-bound-public-access" linkName="Type Bound Public Access" />: is a special <ProductName format={ProductNameFormat.ShortForm}/> concept (represented by `<type>:*`) can be used in relationship tuples to represent every object of that type
```

  ]}
/>

:::caution Wildcard syntax usage
Please note that `*` is a special <ProductName format={ProductNameFormat.ShortForm}/> syntax meaning **everyone** when used as a _user_ within a _relationship tuple_. It is not a wildcard or regex expression.

**You cannot use the `*` syntax in the tuple's object field.**

The following syntax is invalid:
<WriteRequestViewer
  relationshipTuples={[
    {

>       _description: 'Not allow to specify * in the object field',

Reviewer: ```suggestion
      _description: 'It is invalid to use this syntax in the object field. The below relationship tuple is invalid and does not mean that Anne can view all documents.',
```

>       user: 'user:*',

Reviewer: ```suggestion
      user: 'user:anne',
```

>       relation: 'view',

Reviewer: ```suggestion
      relation: 'can_view',
```

Author: We should use relation `view` as I want the example to relate to the model described in the page https://openfga.dev/docs/modeling/public-access

      object: 'document:*',
    },
  ]}
/>

> 

Reviewer: **You cannot use the this syntax as part of a userset in the tuple's user field.**

The following syntax is invalid:
```jsx
<WriteRequestViewer
  relationshipTuples={[
    {
      _description: 'It is invalid to use this syntax as part of a userset. The below relationship tuple is invalid and does not mean that members of any org can view the Getting Started document.',
      user: 'org:*#member',
      relation: 'can_view',
      object: 'document:getting-started',
    },
  ]}
/>
```

Author: Added

  ]}
/>

:::caution Wildcard syntax usage

Please note that type-bound public access is not a wildcard or a regex expression.

**You cannot use the `<type>:*` syntax in the tuple's object field.**

The following syntax is invalid:

<RelationshipTuplesViewer
  relationshipTuples={[
    {
      _description: 'It is invalid to use this syntax in the object field. The below relationship tuple is invalid and does not mean that Bob can view all documents.',
      user: 'user:bob',
      relation: 'view',
      object: 'document:*',
    },
  ]}
/>

:::

:::caution Wildcard syntax usage

**You cannot use `<type>:*` as part of a userset in the tuple's user field.**

The following syntax is invalid:

<RelationshipTuplesViewer
  relationshipTuples={[
    {
      _description: 'It is invalid to use this syntax as part of a userset. The below relationship tuple is invalid and does not mean that members of any org can view the company-psa document.',
      user: 'org:*#member',
      relation: 'view',
      object: 'document:company-psa.doc',
    },
  ]}
/>

:::

### 02. Check That The Relationship Exists

> Once the above _relationship tuple_ is added, we can <ProductConcept section="what-is-a-check-request" linkName="check" /> if **bob** cab `view` `document`:**company-psa.doc**. <ProductName format={ProductNameFormat.ShortForm}/> will return `{ "allowed": true }` even though no relationship tuple linking **bob** to the document was added. That is because the relationship tuple with `*` as the user made it so everyone can `view` the document, making it public.

Reviewer: ```suggestion
Once the above _relationship tuple_ is added, we can <ProductConcept section="what-is-a-check-request" linkName="check" /> if **bob** cab `view` `document`:**company-psa.doc**. <ProductName format={ProductNameFormat.ShortForm}/> will return `{ "allowed": true }` even though no relationship tuple linking **bob** to the document was added. That is because the relationship tuple with `user:*` as the user made it so every object of type user (such as `user:bob`) can `view` the document, making it public.
```
